diff --git a/metadata/pixdecor.xml b/metadata/pixdecor.xml
index 571e4cc..8664dae 100644
--- a/metadata/pixdecor.xml
+++ b/metadata/pixdecor.xml
@@ -282,6 +282,54 @@
 				<default></default>
 				<hint>file</hint>
 			</option>
+			<option name="button_minimize_inactive_image" type="string">
+				<_short>Inactive Button Minimize Image</_short>
+				<_long>The png image to use for the minimize button when the window is not focused.</_long>
+				<default></default>
+				<hint>file</hint>
+			</option>
+			<option name="button_maximize_inactive_image" type="string">
+				<_short>Inactive Button Maximize Image</_short>
+				<_long>The png image to use for the maximize button when the window is not focused.</_long>
+				<default></default>
+				<hint>file</hint>
+			</option>
+			<option name="button_restore_inactive_image" type="string">
+				<_short>Inactive Button Restore Image</_short>
+				<_long>The png image to use for the restore button when the window is not focused.</_long>
+				<default></default>
+				<hint>file</hint>
+			</option>
+			<option name="button_close_inactive_image" type="string">
+				<_short>Inactive Button Close Image</_short>
+				<_long>The png image to use for the close button when the window is not focused.</_long>
+				<default></default>
+				<hint>file</hint>
+			</option>
+			<option name="button_minimize_inactive_hover_image" type="string">
+				<_short>Inactive Button Minimize Hover Image</_short>
+				<_long>The png image to use when hovering the minimize button in a window that is not focused.</_long>
+				<default></default>
+				<hint>file</hint>
+			</option>
+			<option name="button_maximize_inactive_hover_image" type="string">
+				<_short>Inactive Button Maximize Hover Image</_short>
+				<_long>The png image to use when hovering the maximize button in a window that is not focused.</_long>
+				<default></default>
+				<hint>file</hint>
+			</option>
+			<option name="button_restore_inactive_hover_image" type="string">
+				<_short>Inactive Button Restore Hover Image</_short>
+				<_long>The png image to use when hovering the restore button in a window that is not focused.</_long>
+				<default></default>
+				<hint>file</hint>
+			</option>
+			<option name="button_close_inactive_hover_image" type="string">
+				<_short>Inactive Button Close Hover Image</_short>
+				<_long>The png image to use when hovering the close button in a window that is not focused.</_long>
+				<default></default>
+				<hint>file</hint>
+			</option>
 		</group>
 		<group>
 			<_short>Shade</_short>
diff --git a/src/deco-button.cpp b/src/deco-button.cpp
index 3892fdd..6b0b298 100644
--- a/src/deco-button.cpp
+++ b/src/deco-button.cpp
@@ -64,7 +64,7 @@ void button_t::set_pressed(bool is_pressed)
     add_idle_damage();
 }
 
-void button_t::render(const wf::scene::render_instruction_t& data, wf::geometry_t geometry)
+void button_t::render(const wf::scene::render_instruction_t& data, wf::geometry_t geometry, bool active)
 {
     if (this->hover.running())
     {
@@ -72,6 +72,9 @@ void button_t::render(const wf::scene::render_instruction_t& data, wf::geometry_
     }
 
-    OpenGL::render_texture(wf::gles_texture_t{button_texture.get_texture()}, data.target, geometry, {1, 1, 1, this->hover},
+    auto& normal_texture = active ? button_texture : button_texture_inactive;
+    auto& hovered_texture = active ? button_texture_hovered : button_texture_inactive_hovered;
+
+    OpenGL::render_texture(wf::gles_texture_t{normal_texture.get_texture()}, data.target, geometry, {1, 1, 1, this->hover},
         OpenGL::RENDER_FLAG_CACHED);
     data.pass->custom_gles_subpass(data.target,[&]
     {
@@ -84,7 +87,7 @@ void button_t::render(const wf::scene::render_instruction_t& data, wf::geometry_t
     });
     OpenGL::clear_cached();
 
-    OpenGL::render_texture(wf::gles_texture_t{button_texture_hovered.get_texture()}, data.target, geometry, {1, 1, 1, 1.0 - this->hover},
+    OpenGL::render_texture(wf::gles_texture_t{hovered_texture.get_texture()}, data.target, geometry, {1, 1, 1, 1.0 - this->hover},
         OpenGL::RENDER_FLAG_CACHED);
     data.pass->custom_gles_subpass(data.target,[&]
     {
@@ -105,19 +108,25 @@ wf::dimensions_t button_t::update_texture()
         .border = 1.0,
     };
 
-    auto surfaces = theme.get_button_surface(type, state);
-    wf::dimensions_t size_normal{cairo_image_surface_get_width(surfaces->normal), cairo_image_surface_get_height(surfaces->normal)};
-    //wf::dimensions_t size_hovered{cairo_image_surface_get_width(surfaces->hovered), cairo_image_surface_get_height(surfaces->hovered)};
+    auto active_surfaces = theme.get_button_surface(type, state, true);
+    auto inactive_surfaces = theme.get_button_surface(type, state, false);
+    wf::dimensions_t size_normal{cairo_image_surface_get_width(active_surfaces->normal),
+        cairo_image_surface_get_height(active_surfaces->normal)};
 
     wf::gles::run_in_context([&]
     {
-        this->button_texture = owned_texture_t{surfaces->normal};
-        this->button_texture_hovered = owned_texture_t{surfaces->hovered};
+        this->button_texture = owned_texture_t{active_surfaces->normal};
+        this->button_texture_hovered = owned_texture_t{active_surfaces->hovered};
+        this->button_texture_inactive = owned_texture_t{inactive_surfaces->normal};
+        this->button_texture_inactive_hovered = owned_texture_t{inactive_surfaces->hovered};
     });
 
-    cairo_surface_destroy(surfaces->normal);
-    cairo_surface_destroy(surfaces->hovered);
-    surfaces.reset();
+    cairo_surface_destroy(active_surfaces->normal);
+    cairo_surface_destroy(active_surfaces->hovered);
+    cairo_surface_destroy(inactive_surfaces->normal);
+    cairo_surface_destroy(inactive_surfaces->hovered);
+    active_surfaces.reset();
+    inactive_surfaces.reset();
 
     return size_normal;
 }
diff --git a/src/deco-button.hpp b/src/deco-button.hpp
index 68e038a..ca9af4c 100644
--- a/src/deco-button.hpp
+++ b/src/deco-button.hpp
@@ -66,7 +66,7 @@ class button_t
      * @param geometry The geometry of the button, in logical coordinates
      * @param scissor The scissor rectangle to render.
      */
-    void render(const wf::scene::render_instruction_t& data, wf::geometry_t geometry);
+    void render(const wf::scene::render_instruction_t& data, wf::geometry_t geometry, bool active);
 
     pixdecor_theme_t& theme;
     std::function<void()> damage_callback;
@@ -83,7 +83,8 @@ class button_t
     button_type_t type;
     wf::owned_texture_t button_texture;
     wf::owned_texture_t button_texture_hovered;
-    bool active = false;
+    wf::owned_texture_t button_texture_inactive;
+    wf::owned_texture_t button_texture_inactive_hovered;
     wf::geometry_t geometry;
 
     /* Whether the button is currently being hovered */
diff --git a/src/deco-subsurface.cpp b/src/deco-subsurface.cpp
index 85533e4..786fcac 100644
--- a/src/deco-subsurface.cpp
+++ b/src/deco-subsurface.cpp
@@ -205,7 +205,7 @@ class simple_decoration_node_t : public wf::scene::node_t, public wf::pointer_int
                 } else // button
                 {
                     item->as_button().render(data,
-                        item->get_geometry() + origin);
+                        item->get_geometry() + origin, activated);
                 }
             }
         });
diff --git a/src/deco-theme.cpp b/src/deco-theme.cpp
index 7f9050b..029aa09 100644
--- a/src/deco-theme.cpp
+++ b/src/deco-theme.cpp
@@ -23,6 +23,14 @@ wf::option_wrapper_t<std::string> button_minimize_hover_image{"pixdecor/button_mi
 wf::option_wrapper_t<std::string> button_maximize_hover_image{"pixdecor/button_maximize_hover_image"};
 wf::option_wrapper_t<std::string> button_restore_hover_image{"pixdecor/button_restore_hover_image"};
 wf::option_wrapper_t<std::string> button_close_hover_image{"pixdecor/button_close_hover_image"};
+wf::option_wrapper_t<std::string> button_minimize_inactive_image{"pixdecor/button_minimize_inactive_image"};
+wf::option_wrapper_t<std::string> button_maximize_inactive_image{"pixdecor/button_maximize_inactive_image"};
+wf::option_wrapper_t<std::string> button_restore_inactive_image{"pixdecor/button_restore_inactive_image"};
+wf::option_wrapper_t<std::string> button_close_inactive_image{"pixdecor/button_close_inactive_image"};
+wf::option_wrapper_t<std::string> button_minimize_inactive_hover_image{"pixdecor/button_minimize_inactive_hover_image"};
+wf::option_wrapper_t<std::string> button_maximize_inactive_hover_image{"pixdecor/button_maximize_inactive_hover_image"};
+wf::option_wrapper_t<std::string> button_restore_inactive_hover_image{"pixdecor/button_restore_inactive_hover_image"};
+wf::option_wrapper_t<std::string> button_close_inactive_hover_image{"pixdecor/button_close_inactive_hover_image"};
 wf::option_wrapper_t<wf::color_t> button_color{"pixdecor/button_color"};
 wf::option_wrapper_t<double> button_line_thickness{"pixdecor/button_line_thickness"};
 /** Create a new theme with the default parameters */
@@ -327,9 +335,22 @@ static cairo_surface_t *get_cairo_surface(button_type_t button, int w, int h, int
     return surface;
 }
 
-static bool create_button_surfaces(std::unique_ptr<button_surfaces_t>& button_surfaces, std::string button_normal_image, std::string button_hover_image)
+static bool create_button_surfaces(std::unique_ptr<button_surfaces_t>& button_surfaces,
+    std::string button_normal_image, std::string button_hover_image,
+    std::string button_inactive_image, std::string button_inactive_hover_image, bool active)
 {
     bool normal_same_as_hover = false;
+    if (!active)
+    {
+        if (!button_inactive_image.empty())
+        {
+            button_normal_image = button_inactive_image;
+        }
+        if (!button_inactive_hover_image.empty())
+        {
+            button_hover_image = button_inactive_hover_image;
+        }
+    }
     if (!button_normal_image.empty())
     {
         button_surfaces->normal = cairo_image_surface_create_from_png(button_normal_image.c_str());
@@ -350,29 +369,37 @@ static bool create_button_surfaces(std::unique_ptr<button_surfaces_t>& button_sur
 }
 
 std::unique_ptr<button_surfaces_t> pixdecor_theme_t::get_button_surface(button_type_t button,
-    const button_state_t& state) const
+    const button_state_t& state, bool active) const
 {
     std::unique_ptr<button_surfaces_t> button_surfaces = std::make_unique<button_surfaces_t>();
     bool normal_same_as_hover = false;
 
     switch (button)
     {
       case BUTTON_CLOSE:
-        normal_same_as_hover = create_button_surfaces(button_surfaces, std::string(button_close_image), std::string(button_close_hover_image));
+        normal_same_as_hover = create_button_surfaces(button_surfaces,
+            std::string(button_close_image), std::string(button_close_hover_image),
+            std::string(button_close_inactive_image), std::string(button_close_inactive_hover_image), active);
         break;
 
       case BUTTON_TOGGLE_MAXIMIZE:
         if (this->maximized)
         {
-            normal_same_as_hover = create_button_surfaces(button_surfaces, std::string(button_restore_image), std::string(button_restore_hover_image));
+            normal_same_as_hover = create_button_surfaces(button_surfaces,
+                std::string(button_restore_image), std::string(button_restore_hover_image),
+                std::string(button_restore_inactive_image), std::string(button_restore_inactive_hover_image), active);
         } else
         {
-            normal_same_as_hover = create_button_surfaces(button_surfaces, std::string(button_maximize_image), std::string(button_maximize_hover_image));
+            normal_same_as_hover = create_button_surfaces(button_surfaces,
+                std::string(button_maximize_image), std::string(button_maximize_hover_image),
+                std::string(button_maximize_inactive_image), std::string(button_maximize_inactive_hover_image), active);
         }
         break;
 
       case BUTTON_MINIMIZE:
-        normal_same_as_hover = create_button_surfaces(button_surfaces, std::string(button_minimize_image), std::string(button_minimize_hover_image));
+        normal_same_as_hover = create_button_surfaces(button_surfaces,
+            std::string(button_minimize_image), std::string(button_minimize_hover_image),
+            std::string(button_minimize_inactive_image), std::string(button_minimize_inactive_hover_image), active);
         break;
 
       default:
diff --git a/src/deco-theme.hpp b/src/deco-theme.hpp
index 85b37ca..b60e421 100644
--- a/src/deco-theme.hpp
+++ b/src/deco-theme.hpp
@@ -79,7 +79,7 @@ class pixdecor_theme_t
      * @param state The button state.
      */
     std::unique_ptr<button_surfaces_t> get_button_surface(button_type_t button,
-        const button_state_t& state) const;
+        const button_state_t& state, bool active) const;
 
     void set_maximize(bool state);
 
diff --git a/src/decoration.cpp b/src/decoration.cpp
index db22493..370b2a1 100644
--- a/src/decoration.cpp
+++ b/src/decoration.cpp
@@ -73,6 +73,14 @@ class wayfire_pixdecor : public wf::plugin_interface_t
     wf::option_wrapper_t<std::string> button_maximize_hover_image{"pixdecor/button_maximize_hover_image"};
     wf::option_wrapper_t<std::string> button_restore_hover_image{"pixdecor/button_restore_hover_image"};
     wf::option_wrapper_t<std::string> button_close_hover_image{"pixdecor/button_close_hover_image"};
+    wf::option_wrapper_t<std::string> button_minimize_inactive_image{"pixdecor/button_minimize_inactive_image"};
+    wf::option_wrapper_t<std::string> button_maximize_inactive_image{"pixdecor/button_maximize_inactive_image"};
+    wf::option_wrapper_t<std::string> button_restore_inactive_image{"pixdecor/button_restore_inactive_image"};
+    wf::option_wrapper_t<std::string> button_close_inactive_image{"pixdecor/button_close_inactive_image"};
+    wf::option_wrapper_t<std::string> button_minimize_inactive_hover_image{"pixdecor/button_minimize_inactive_hover_image"};
+    wf::option_wrapper_t<std::string> button_maximize_inactive_hover_image{"pixdecor/button_maximize_inactive_hover_image"};
+    wf::option_wrapper_t<std::string> button_restore_inactive_hover_image{"pixdecor/button_restore_inactive_hover_image"};
+    wf::option_wrapper_t<std::string> button_close_inactive_hover_image{"pixdecor/button_close_inactive_hover_image"};
     wf::option_wrapper_t<std::string> ignore_views_string{"pixdecor/ignore_views"};
     wf::option_wrapper_t<std::string> always_decorate_string{"pixdecor/always_decorate"};
     wf::option_wrapper_t<std::string> effect_type{"pixdecor/effect_type"};
@@ -440,6 +448,14 @@ class wayfire_pixdecor : public wf::plugin_interface_t
         button_maximize_hover_image.set_callback([=] {recreate_frames();});
         button_restore_hover_image.set_callback([=] {recreate_frames();});
         button_close_hover_image.set_callback([=] {recreate_frames();});
+        button_minimize_inactive_image.set_callback([=] {recreate_frames();});
+        button_maximize_inactive_image.set_callback([=] {recreate_frames();});
+        button_restore_inactive_image.set_callback([=] {recreate_frames();});
+        button_close_inactive_image.set_callback([=] {recreate_frames();});
+        button_minimize_inactive_hover_image.set_callback([=] {recreate_frames();});
+        button_maximize_inactive_hover_image.set_callback([=] {recreate_frames();});
+        button_restore_inactive_hover_image.set_callback([=] {recreate_frames();});
+        button_close_inactive_hover_image.set_callback([=] {recreate_frames();});
         title_text_align.set_callback([=]
         {
             for (auto& view : wf::get_core().get_all_views())
